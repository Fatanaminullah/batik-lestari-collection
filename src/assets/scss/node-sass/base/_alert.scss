// Alert base
// Version 1.1.0
//
// Custom styles for alert base.

.alert {
  display: flex;
  gap: 0.75rem;
  align-items: flex-start;
  max-width: 500px;

  .close {
    position: absolute;
    color: $white;
    text-shadow: none;
    @include transition($transition-base);
    @include hover-focus-active {
      color: $white !important;
    }
  }

  .alert-title {
    font-weight: bold;
  }

  .alert-text {
    font-size: 14px;
  }

  &.alert-filled {
    color: $white;

    &.alert-primary {
      border-color: $primary;
      background-color: $primary;
    }

    &.alert-secondary {
      border-color: $secondary;
      background-color: $secondary;
    }

    &.alert-info {
      border-color: $info;
      background-color: $info;
    }

    &.alert-danger {
      border-color: $danger;
      background-color: $danger;
    }

    &.alert-success {
      border-color: $success;
      background-color: $success;
    }
  }

  &.alert-boxless {
    background-color: transparent;
    border-color: transparent;
  }

  &:not(.alert-filled) {
    .alert-text {
      color: rgba(26, 26, 26, 0.6);
    }
  }
}
