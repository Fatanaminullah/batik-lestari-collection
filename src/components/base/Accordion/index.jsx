// ░█▀▀▄ ░█▀▀▀█ 　 ░█▄─░█ ░█▀▀▀█ ▀▀█▀▀ 　 ░█▀▀▀ ░█▀▀▄ ▀█▀ ▀▀█▀▀
// ░█─░█ ░█──░█ 　 ░█░█░█ ░█──░█ ─░█── 　 ░█▀▀▀ ░█─░█ ░█─ ─░█──
// ░█▄▄▀ ░█▄▄▄█ 　 ░█──▀█ ░█▄▄▄█ ─░█── 　 ░█▄▄▄ ░█▄▄▀ ▄█▄ ─░█──
// The component is made to be used flexibly. But if you REALLY MUST update edit the component,
// please add `-custom` on the change log file (e.g. `1.0.0-custom.md`) and log the changes

// ========================================================================================================================================
// Accordion
// ========================================================================================================================================

import React from "react"
import PropTypes from "prop-types"

import {
  Accordion as AccordionMaterial,
  AccordionDetails,
  AccordionSummary,
} from "@mui/material"

export function Accordion({
  forwardRef,
  id,
  variant,
  expandedKey,
  expandIcon,
  expandIconPlus,
  title,
  titleClassName,
  className,
  headerClassName,
  contentClassName,
  defaultExpanded,
  children,
  expanded,
  onChange,
}) {
  const accordionVariant = () => `accordion-${variant}`

  return (
    <AccordionMaterial
      className={`accordion ${className} ${accordionVariant()}`}
      id={id}
      ref={forwardRef}
      defaultExpanded={defaultExpanded === expandedKey}
      expandedKey={expandedKey}
      expanded={expanded}
      onChange={onChange}
    >
      <AccordionSummary
        expandIcon={expandIcon}
        className={`accordion-header ${headerClassName} ${
          expandIconPlus && "accordion-header-plus"
        }`}
      >
        <h5 className={`accordion-title ${titleClassName}`}>{title}</h5>
        {expandIconPlus && (
          <div className="expand-plus">
            <div className="h-bar" />
            <div className="v-bar" />
          </div>
        )}
      </AccordionSummary>
      <AccordionDetails className={`accordion-content ${contentClassName}`}>
        {children}
      </AccordionDetails>
    </AccordionMaterial>
  )
}

Accordion.propTypes = {
  forwardRef: PropTypes.any,
  // expandKey: PropTypes.any,
  expandIcon: PropTypes.any,
  expandIconPlus: PropTypes.bool,
  id: PropTypes.string,
  variant: PropTypes.oneOf(["basic", "line"]),
  // onClick: PropTypes.any,
  children: PropTypes.any,
  title: PropTypes.any,
  titleClassName: PropTypes.string,
  className: PropTypes.string,
  headerClassName: PropTypes.string,
  contentClassName: PropTypes.string,
}

Accordion.defaultProps = {
  expandIcon: <i className="ics ic-chevron-down" />,
  variant: "basic",
  title: "Title",
  titleClassName: "",
  className: "",
  headerClassName: "",
  contentClassName: "",
}
