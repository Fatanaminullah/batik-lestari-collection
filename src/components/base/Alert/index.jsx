import React from "react"
import PropTypes from "prop-types"

export function Alert({ variant, type, title, description, showIcon }) {
  return (
    <div className={`alert alert-${variant} alert-${type}`}>
      {showIcon ? (
        <div className="alert-icon">
          <i className="ics ic-check-circle" />
        </div>
      ) : null}
      <div>
        <div className="alert-title">{title}</div>
        <div className="alert-text">{description}</div>
      </div>
    </div>
  )
}

Alert.defaultProps = {
  variant: "info",
  type: "outline",
  showIcon: true,
  title: "Title Message",
  description:
    "Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore.",
}

Alert.propTypes = {
  variant: PropTypes.oneOf([
    "primary",
    "secondary",
    "info",
    "danger",
    "success",
    "link",
    "dark",
    "light",
  ]),
  type: PropTypes.oneOf(["outline", "filled", "boxless"]),
  title: PropTypes.string,
  description: PropTypes.string,
  showIcon: PropTypes.bool,
}
