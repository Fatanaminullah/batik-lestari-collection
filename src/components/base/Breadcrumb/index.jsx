import PropTypes from "prop-types"
import { useRouter } from "next/router"
import { Link } from "../Link"

export function Breadcrumb({ id, path, theme, className }) {
  const router = useRouter()
  const isObjectPath = typeof path === "object"
  const defaultPath = path ?? router.pathname

  return (
    <>
      <ol id={id} className={`breadcrumb ${theme} ${className}`}>
        <li className="breadcrumb-item" key="home-path">
          <Link href="/">Home</Link>
        </li>
        {isObjectPath ? (
          <>
            {path.map((item, i) => (
              <li key={i} className="breadcrumb-item">
                {i + 1 === path.length ? (
                  <span>{item.title}</span>
                ) : (
                  <Link href={`${item.url}`}>{item.title}</Link>
                )}
              </li>
            ))}
          </>
        ) : (
          <>
            {defaultPath
              .split("/")
              .slice(1)
              .map((item, i) => (
                <li key={i} className="breadcrumb-item">
                  {i + 1 === defaultPath.split().length ? (
                    <span>{item}</span>
                  ) : (
                    <Link href={`${item}`}>{item}</Link>
                  )}
                </li>
              ))}
          </>
        )}
      </ol>
    </>
  )
}

Breadcrumb.propTypes = {
  id: PropTypes.string,
  path: PropTypes.oneOfType([PropTypes.object, PropTypes.string]),
  className: PropTypes.string,
  theme: PropTypes.oneOf(["breadcrumb-light", "breadcrumb-dark"]),
}

Breadcrumb.defaultProps = {
  className: "",
  theme: "breadcrumb-light",
}

// const stringPath = "/works/about"
// const customPath = [{ title: "About", url: '/about' }]
// const default = useRouter().pathname
