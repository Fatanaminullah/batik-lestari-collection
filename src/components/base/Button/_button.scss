// ░█▀▀▄ ░█▀▀▀█ 　 ░█▄─░█ ░█▀▀▀█ ▀▀█▀▀ 　 ░█▀▀▀ ░█▀▀▄ ▀█▀ ▀▀█▀▀
// ░█─░█ ░█──░█ 　 ░█░█░█ ░█──░█ ─░█── 　 ░█▀▀▀ ░█─░█ ░█─ ─░█──
// ░█▄▄▀ ░█▄▄▄█ 　 ░█──▀█ ░█▄▄▄█ ─░█── 　 ░█▄▄▄ ░█▄▄▀ ▄█▄ ─░█──
// Override style on .buttons-custom.scss instead
// unless you want to upgrade the component version

// ========================================================================================================================================
// Buttons
// ========================================================================================================================================

button {
  &:disabled {
    cursor: default !important;
  }
}

.btn {
  position: relative;
  text-transform: $btn-text-transform;
  letter-spacing: $btn-letter-spacing;

  .el-loader {
    display: none;
  }
  &.loading,
  &.success {
    span,
    &:after,
    &:before {
      visibility: hidden;
    }
    .el-loader {
      display: block;

      &:before,
      &:after {
        height: 1rem;
        width: 1rem;
        line-height: 1rem;
        top: 50%;
        margin-top: -0.5rem;
        left: 50%;
        margin-left: -0.5rem;
        position: absolute;
        font-style: normal;
      }
    }
  }
  &.loading {
    pointer-events: none;
    .el-loader {
      animation: fadeIn 0.2s linear;
      &:before {
        font-family: "Base Icons Regular";
        content: varIcon($ic-circle);
        opacity: 0.4;
        z-index: 1;
        font-size: 1rem;
      }
      &:after {
        font-family: "Base Icons Regular";
        content: varIcon($ic-spinner-third);
        animation: spin 2s infinite linear;
        z-index: 2;
        font-size: 1rem;
      }
    }
  }
  &.success {
    pointer-events: none;
    .el-loader {
      &:after {
        font-family: "Base Icons Regular";
        content: varIcon($ic-check);
        animation: flash 1s linear;
      }
    }
  }
}

// Button link
.btn-link {
  padding-left: 0;
  padding-right: 0;
  &:focus,
  &:active {
    box-shadow: none;
  }
}

// Button Light / White
.btn-light,
.btn-white {
  color: $primary;
  @include hover-focus-active {
    color: $primary;
  }
}

// Button group
// Set display block to make .text-* on parent element / its own element
// affects the button children element
.btn-group {
  display: block;
}

.btn {
  &.btn-ic-left {
    padding-left: 2.5rem;
  }
  &.btn-ic-right {
    padding-right: 2.5rem;
  }
  .el-ic-before {
    &:before {
      top: 50%;
      transform: translateY(-50%);
      left: 1rem;
      position: absolute;
    }
  }
  .el-ic-after {
    &:before {
      top: 50%;
      transform: translateY(-50%);
      right: 1rem;
      position: absolute;
    }
  }
  &.btn-link {
    .el-ic-before {
      &:before {
        left: 0;
        transition: $transition-base;
      }
    }
    .el-ic-after {
      &:before {
        right: 0;
        transition: $transition-base;
      }
    }
    &.btn-ic-left {
      padding-left: 1rem;
      &:before {
        display: none;
      }
      &:hover {
        .el-ic-before {
          &:before {
            left: -0.25rem;
          }
        }
      }
    }
    &.btn-ic-right {
      padding-right: 1rem;
      &:before {
        display: none;
      }
      &:hover {
        .el-ic-after {
          &:before {
            right: -0.25rem;
          }
        }
      }
    }
  }
}
