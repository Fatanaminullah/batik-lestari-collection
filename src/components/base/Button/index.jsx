// ░█▀▀▄ ░█▀▀▀█ 　 ░█▄─░█ ░█▀▀▀█ ▀▀█▀▀ 　 ░█▀▀▀ ░█▀▀▄ ▀█▀ ▀▀█▀▀
// ░█─░█ ░█──░█ 　 ░█░█░█ ░█──░█ ─░█── 　 ░█▀▀▀ ░█─░█ ░█─ ─░█──
// ░█▄▄▀ ░█▄▄▄█ 　 ░█──▀█ ░█▄▄▄█ ─░█── 　 ░█▄▄▄ ░█▄▄▀ ▄█▄ ─░█──
// The component is made to be used flexibly. But if you REALLY MUST update edit the component,
// please add `-custom` on the change log file (e.g. `1.0.0-custom.md`) and log the changes

// ========================================================================================================================================
// Buttons
// ========================================================================================================================================

import { useMemo } from "react"
import PropTypes from "prop-types"
import { Link } from "../Link"

// ==========================================================================
// Button
// ==========================================================================

export function Button({
  id,
  variant,
  iconLeft,
  iconRight,
  link,
  target,
  size,
  className,
  onClick,
  children,
  disabled,
  type,
}) {
  const btnSize = useMemo(() => {
    if (!(size === "")) {
      return ` btn-${size}`
    }

    return ""
  }, [size])

  const btnVariant = () => ` btn-${variant}`

  // prettier-ignore
  if (link) {
    return (
      <Link
        href={link}
        target={target}
        className={`btn${btnVariant()}${btnSize}${iconLeft ? " btn-ic-left" : ""}${iconRight ? " btn-ic-right" : ""} ${className}`}
        id={id}
        onClick={onClick}
        disabled={disabled}
      >
        <span>
          {typeof iconLeft === "string" && <i className={`${iconLeft} el-ic-before`} />}
          {children}
          {typeof iconRight === "string" && <i className={`${iconRight} el-ic-after`} />}
        </span>
        <div className="el-block" />
        <div className="el-loader" />
      </Link>
    )
  }

  return (
    <button
      type={type}
      className={`btn${btnVariant()}${btnSize}${
        iconLeft ? " btn-ic-left" : ""
      }${iconRight ? " btn-ic-right" : ""} ${className}`}
      id={id}
      onClick={onClick}
      disabled={disabled}
    >
      <span>
        {typeof iconLeft === "string" && (
          <i className={`${iconLeft} el-ic-before`} />
        )}
        {children}
        {typeof iconRight === "string" && (
          <i className={`${iconRight} el-ic-after`} />
        )}
      </span>
      <div className="el-block" />
      <div className="el-loader" />
    </button>
  )
}

Button.propTypes = {
  variant: PropTypes.oneOf([
    "primary",
    "secondary",
    "outline-primary",
    "outline-secondary",
    "info",
    "outline-info",
    "danger",
    "outline-danger",
    "success",
    "outline-success",
    "link",
    "dark",
    "light",
    "outline-white",
  ]), // Set variant (based on Bootstrap classes)
  iconLeft: PropTypes.string,
  iconRight: PropTypes.string,
  link: PropTypes.string, // Set button link / href
  target: PropTypes.string,
  size: PropTypes.oneOf(["sm", "lg", "md"]), // Set button size
  className: PropTypes.string,
  id: PropTypes.string,
  onClick: PropTypes.func,
  children: PropTypes.any,
  type: PropTypes.string,
  disabled: PropTypes.bool,
}

Button.defaultProps = {
  variant: "primary",
  target: null,
  size: "md",
  className: "",
  type: "button",
  disabled: false,
}
