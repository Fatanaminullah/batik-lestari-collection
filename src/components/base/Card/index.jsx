// ░█▀▀▄ ░█▀▀▀█ 　 ░█▄─░█ ░█▀▀▀█ ▀▀█▀▀ 　 ░█▀▀▀ ░█▀▀▄ ▀█▀ ▀▀█▀▀
// ░█─░█ ░█──░█ 　 ░█░█░█ ░█──░█ ─░█── 　 ░█▀▀▀ ░█─░█ ░█─ ─░█──
// ░█▄▄▀ ░█▄▄▄█ 　 ░█──▀█ ░█▄▄▄█ ─░█── 　 ░█▄▄▄ ░█▄▄▀ ▄█▄ ─░█──
// The component is made to be used flexibly. But if you REALLY MUST update edit the component,
// please add `-custom` on the change log file (e.g. `1.0.0-custom.md`) and log the changes

// ========================================================================================================================================
// Card
// ========================================================================================================================================

import PropTypes from "prop-types"
import imgDummy from "@public/img/dummy/img_dummy.jpg"
import { RatioImage, Image } from "../Image"
import { Link } from "../Link"

export function Card({
  id,
  forwardRef,
  variant,
  colLeft,
  colRight,
  label,
  labelClassName,
  title,
  titleClassName,
  text,
  textClassName,
  icon,
  cardIconClassName,
  iconWidth,
  img,
  imgHeight,
  imgRatio,
  imgAlt,
  imgOverlay,
  link,
  target,
  cardBgClassName,
  cardBodyClassName,
  className,
  children,
  onClick,
}) {
  const cardVariant = () => `card-${variant}`

  return (
    <Link
      href={link}
      className={`card ${cardVariant()} ${className}`}
      target={target}
      id={id}
      ref={forwardRef}
      onClick={onClick}
    >
      <div className="row row-0">
        {icon ||
          (img && (
            <div className={`col-bg ${colLeft}`}>
              {icon && (
                <div className={`card-icon ${cardIconClassName}`}>
                  <Image
                    src={icon}
                    className={`img-fluid card-icon-img ${iconWidth}`}
                    alt={title || "Icon"}
                  />
                </div>
              )}
              {img && (
                <div className={`card-bg ${cardBgClassName}`}>
                  <div className={`card-bg-img ${imgHeight}`}>
                    <div className={`bg-overlay bg-overlay-${imgOverlay}`} />
                    <RatioImage src={img} ratio={imgRatio} alt={imgAlt} fill />
                  </div>
                </div>
              )}
            </div>
          ))}
        <div className={`col-body ${colRight}`}>
          {(title || text || children) && (
            <div className={`card-body ${cardBodyClassName}`}>
              {label && (
                <span className={`card-label ${labelClassName}`}>{label}</span>
              )}
              {title && (
                <h4 className={`card-title ${titleClassName}`}>{title}</h4>
              )}
              {text && <p className={`card-text ${textClassName}`}>{text}</p>}
              {children}
            </div>
          )}
        </div>
      </div>
    </Link>
  )
}

Card.propTypes = {
  variant: PropTypes.oneOf(["outline", "elevated", "boxless", "overlay"]), // Set card variant
  colLeft: PropTypes.string, // Set col left utilities classes (e.g. .col-md-6, .col-lg-4, etc.)
  colRight: PropTypes.string, // Set col right utilities classes (e.g. .col-md-6, .col-lg-8, etc.)
  label: PropTypes.string,
  labelClassName: PropTypes.string,
  title: PropTypes.string,
  titleClassName: PropTypes.string,
  text: PropTypes.string,
  textClassName: PropTypes.string,
  img: PropTypes.any,
  imgHeight: PropTypes.string,
  imgRatio: PropTypes.string,
  imgAlt: PropTypes.string,
  imgOverlay: PropTypes.string, // Set cover overlay %
  link: PropTypes.string,
  target: PropTypes.string,
  cardBgClassName: PropTypes.string,
  cardBodyClassName: PropTypes.string,
  className: PropTypes.string,
  id: PropTypes.string,
  onClick: PropTypes.any,
  children: PropTypes.any,
}

Card.defaultProps = {
  variant: "basic",
  colLeft: "col-12",
  colRight: "col-12",
  label: "Label",
  labelClassName: "",
  title: "Title",
  titleClassName: "",
  text: "Text",
  textClassName: "",
  cardBgClassName: "",
  cardBodyClassName: "",
  className: "",
  imgHeight: "",
  imgRatio: "r-16-9",
  imgOverlay: "bottom-content-wrapper",
  link: "",
  target: null,
  img: imgDummy,
}
