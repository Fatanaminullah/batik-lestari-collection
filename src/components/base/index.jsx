export * from "./Button"
export * from "./Alert"
export * from "./Toast"
export * from "./Slider/Carousel"
export * from "./Cover"
export * from "./Card"
