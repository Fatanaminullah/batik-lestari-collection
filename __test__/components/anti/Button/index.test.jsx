import { render } from "@testing-library/react"
import { Button } from "../../../../src/components/base/Button"

describe("Button component", () => {
  test("it renders correctly with default props", () => {
    const { container } = render(<Button>Default</Button>)
    expect(container).toMatchSnapshot()
  })

  test("it renders correctly with custom variant", () => {
    const { container } = render(<Button variant="secondary">Secondary</Button>)
    expect(container).toMatchSnapshot()
  })

  test("it renders correctly with custom iconLeft", () => {
    const { container } = render(
      <Button variant="secondary" iconLeft="fas fa-user">
        User
      </Button>
    )
    expect(container).toMatchSnapshot()
  })

  test("it renders correctly with custom iconRight", () => {
    const { container } = render(
      <Button variant="secondary" iconRight="fas fa-user">
        User
      </Button>
    )
    expect(container).toMatchSnapshot()
  })

  test("it renders correctly with custom link", () => {
    const { container } = render(
      <Button link="https://google.com">Google</Button>
    )
    expect(container).toMatchSnapshot()
  })

  test("it renders correctly with custom target", () => {
    const { container } = render(
      <Button link="https://google.com" target="_blank">
        Google
      </Button>
    )
    expect(container).toMatchSnapshot()
  })

  test("it renders correctly with custom size", () => {
    const { container } = render(<Button size="lg">Large</Button>)
    expect(container).toMatchSnapshot()
  })

  test("it renders correctly with custom className", () => {
    const { container } = render(
      <Button className="custom-class">Custom</Button>
    )
    expect(container).toMatchSnapshot()
  })

  test("it renders correctly with custom onClick", () => {
    const onClick = jest.fn()
    const { container } = render(<Button onClick={onClick}>Click</Button>)
    expect(container).toMatchSnapshot()
  })

  test("it renders correctly with custom type", () => {
    const { container } = render(<Button type="submit">Submit</Button>)
    expect(container).toMatchSnapshot()
  })

  test("it renders correctly with custom id", () => {
    const { container } = render(<Button id="custom-id">Custom ID</Button>)
    expect(container).toMatchSnapshot()
  })
})
