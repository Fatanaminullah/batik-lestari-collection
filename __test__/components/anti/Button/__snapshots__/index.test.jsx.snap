// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button component it renders correctly with custom className 1`] = `
<div>
  <button
    class="btn btn-primary custom-class"
    type="button"
  >
    <span>
      Custom
    </span>
    <div
      class="el-block"
    />
    <div
      class="el-loader"
    />
  </button>
</div>
`;

exports[`Button component it renders correctly with custom iconLeft 1`] = `
<div>
  <button
    class="btn btn-secondary btn-ic-left "
    type="button"
  >
    <span>
      <i
        class="fas fa-user el-ic-before"
      />
      User
    </span>
    <div
      class="el-block"
    />
    <div
      class="el-loader"
    />
  </button>
</div>
`;

exports[`Button component it renders correctly with custom iconRight 1`] = `
<div>
  <button
    class="btn btn-secondary btn-ic-right "
    type="button"
  >
    <span>
      User
      <i
        class="fas fa-user el-ic-after"
      />
    </span>
    <div
      class="el-block"
    />
    <div
      class="el-loader"
    />
  </button>
</div>
`;

exports[`Button component it renders correctly with custom id 1`] = `
<div>
  <button
    class="btn btn-primary "
    id="custom-id"
    type="button"
  >
    <span>
      Custom ID
    </span>
    <div
      class="el-block"
    />
    <div
      class="el-loader"
    />
  </button>
</div>
`;

exports[`Button component it renders correctly with custom link 1`] = `
<div>
  <a
    class="btn btn-primary "
    href="https://google.com"
    rel="noopener noreferrer"
  >
    <span>
      Google
    </span>
    <div
      class="el-block"
    />
    <div
      class="el-loader"
    />
  </a>
</div>
`;

exports[`Button component it renders correctly with custom onClick 1`] = `
<div>
  <button
    class="btn btn-primary "
    type="button"
  >
    <span>
      Click
    </span>
    <div
      class="el-block"
    />
    <div
      class="el-loader"
    />
  </button>
</div>
`;

exports[`Button component it renders correctly with custom size 1`] = `
<div>
  <button
    class="btn btn-primary btn-lg "
    type="button"
  >
    <span>
      Large
    </span>
    <div
      class="el-block"
    />
    <div
      class="el-loader"
    />
  </button>
</div>
`;

exports[`Button component it renders correctly with custom target 1`] = `
<div>
  <a
    class="btn btn-primary "
    href="https://google.com"
    rel="noopener noreferrer"
    target="_blank"
  >
    <span>
      Google
    </span>
    <div
      class="el-block"
    />
    <div
      class="el-loader"
    />
  </a>
</div>
`;

exports[`Button component it renders correctly with custom type 1`] = `
<div>
  <button
    class="btn btn-primary "
    type="submit"
  >
    <span>
      Submit
    </span>
    <div
      class="el-block"
    />
    <div
      class="el-loader"
    />
  </button>
</div>
`;

exports[`Button component it renders correctly with custom variant 1`] = `
<div>
  <button
    class="btn btn-secondary "
    type="button"
  >
    <span>
      Secondary
    </span>
    <div
      class="el-block"
    />
    <div
      class="el-loader"
    />
  </button>
</div>
`;

exports[`Button component it renders correctly with default props 1`] = `
<div>
  <button
    class="btn btn-primary "
    type="button"
  >
    <span>
      Default
    </span>
    <div
      class="el-block"
    />
    <div
      class="el-loader"
    />
  </button>
</div>
`;
